app/models/pawn.rb

```diff
@@ -1,2 +1,42 @@
 class Pawn < Piece
+  def valid_pawn_move?(to_x:, to_y:)
+    return true if direction_of_white_and_black_pawns(to_y) &&
+      (
+        normal_move?(to_x, to_y) ||
+        first_move?(to_x, to_y) ||
+        pawn_diagonal?(to_x, to_y)
+      )
+    false
+  end
+
+  private
+
+  # This logic makes sure that the pawn can only move forward
+  # White pawns can only move *up* the board
+  # Black pawns can only move *down* the board
+  def direction_of_white_and_black_pawns(to_y)
+    return true if color == 'white' && to_y > y_position
+    return true if color == 'black' && to_y < y_position
+    false
+  end
+
+  # Returns true if the pawn advances one space from its origin
```

Comment:
Totally get it... changed comment to say `# Returns true if the pawn advances one space 
`. Because the `normal_move?` method and the `first_move?` methods can be true at the same time (in the case that the pawn advances a single space on its first move), I'm going to leave it at that.